pydantic/json_schema.py

```diff
@@ -0,0 +1,1081 @@
+    'JsonSchemaMetadata',
+    {
+        # ### "Pre-processing" of the JSON schema
+        # If not None, this CoreSchema will be used to generate the JSON schema instead of the "real one"
+        # You can use a callable to defer evaluation of the CoreSchema until it's needed
+        'core_schema_override': Union[CoreSchema, Callable[[], CoreSchema], None],
+        # A reference to the source class if appropriate; useful when working with some of the plain function schemas
+        'source_class': Union[Type[Any], None],
+        # ### "Miscellaneous properties" that are available for all JSON types
+        # (see https://json-schema.org/understanding-json-schema/reference/generic.html)
+        'title': Optional[str],
+        'description': Optional[str],
+        'examples': Optional[List[Any]],
+        'deprecated': Optional[bool],
+        'readOnly': Optional[bool],
+        'writeOnly': Optional[bool],
+        '$comment': Optional[str],
+        # Note: 'default', which is included with these fields in the JSON Schema docs, is handled by CoreSchema
+        # ### "Post-processing" of the JSON schema
+        # A catch-all for arbitrary data to add to the schema
+        'extra_updates': Optional[Dict[str, Any]],
+        # A final function to apply to the JSON schema after all other modifications have been applied
+        # If you want to force specific contents in the generated schema, you can use a function that ignores the
+        # input value and just return the schema you want.
+        'modify_js_function': Optional[Callable[[JsonSchemaValue], None]],
+    },
+    total=False,
+)
+
+
+def merge_js_metadata(
+    base: JsonSchemaMetadata | None, overrides: JsonSchemaMetadata | None
+) -> JsonSchemaMetadata | None:
+    """
+    Merge two JsonSchemaMetadata dicts, with the values from the second overriding the first.
+
+    Returns a new object, or None if both arguments are None. The provided objects are not modified.
+    """
+    if base is None:
+        if overrides is None:
+            return None
+        else:
+            return overrides.copy()
+    elif overrides is None:
+        return base.copy()
+    else:
+        result: JsonSchemaMetadata = base.copy()
+        result.update(**overrides)  # type: ignore[misc]  # seems like a mypy bug
+        return result
+
+
+def without_core_schema_override(json_schema_metadata: JsonSchemaMetadata) -> JsonSchemaMetadata:
+    """
+    Return a copy without the core_schema_override; when handling the core_schema_override,
+    we need to remove it from the object so that the handler doesn't recurse infinitely.
+    """
+    result = json_schema_metadata.copy()
+    result.pop('core_schema_override', None)
+    return result
+
+
+_STANDARD_FIELDS: list[
+    Literal['title', 'description', 'examples', 'deprecated', 'readOnly', 'writeOnly', '$comment']
+] = ['title', 'description', 'examples', 'deprecated', 'readOnly', 'writeOnly', '$comment']
```

Comment:
```suggestion
_STANDARD_FIELDS_TYPES = Literal['title', 'description', 'examples', 'deprecated', 'readOnly', 'writeOnly', '$comment']
_STANDARD_FIELDS: list[_STANDARD_FIELDS_TYPES] = list(_STANDARD_FIELDS_TYPES.__args__)
```

???

Reply:
You get a mypy error trying to access `__args__` but I think we can use `_STANDARD_FIELDS = all_literal_values(_STANDARD_FIELD_TYPES)`. I'll make the change

Reply:
ok.